ui/redux/selectors.js

```diff
@@ -65,94 +64,90 @@ export const getFamiliesGroupedByProjectGuid = createSelector(getFamiliesByGuid,
-    }), {})
+    return Object.values(individualsByGuid).sort((a, b) => {
+      const compareVal = getIndivAffectedSort(a) - getIndivAffectedSort(b)
+      if (compareVal === 0) {
+        return getIndivMmeSort(b).localeCompare(getIndivMmeSort(a))
+      }
+      return compareVal
+    })
   },
 )
 
-export const getFirstSampleByFamily = createSelector(
-  getFamiliesByGuid,
-  getIndividualsByGuid,
+export const getSortedIndividualsByFamily = createSelector(
+  getSortedIndividuals,
+  groupIndividualsByFamilyGuid,
+)
+
+const getSortedSamples = createSelector(
   getSamplesByGuid,
-  (familiesByGuid, individualsByGuid, samplesByGuid) => {
-    return Object.entries(familiesByGuid).reduce((acc, [familyGuid, family]) => {
-      const familySamples = familyVariantSamples(family, individualsByGuid, samplesByGuid)
+  samplesByGuid => Object.values(samplesByGuid).sort((a, b) => a.loadedDate.localeCompare(b.loadedDate)),
+)
 
-      return {
-        ...acc,
-        [familyGuid]: familySamples.length > 0 ? familySamples[0] : null,
+export const getSamplesByFamily = createSelector(
+  getIndividualsByGuid,
+  getSortedSamples,
+  (individualsByGuid, sortedSamples) =>
+    sortedSamples.reduce((acc, sample) => {
+      const { familyGuid } = individualsByGuid[sample.individualGuid]
+      if (!acc[familyGuid]) {
+        acc[familyGuid] = []
       }
-    }, {})
-  },
+      acc[familyGuid].push(sample)
+      return acc
+    }, {}),
 )
 
 export const getHasActiveVariantSampleByFamily = createSelector(
-  getSortedIndividualsByFamily,
-  getSamplesByGuid,
-  (individualsByFamily, samplesByGuid) => {
-    return Object.entries(individualsByFamily).reduce((acc, [familyGuid, individuals]) => ({
+  getSamplesByFamily,
+  (samplesByFamily) => {
+    return Object.entries(samplesByFamily).reduce((acc, [familyGuid, familySamples]) => ({
       ...acc,
-      [familyGuid]: individuals.some(individual => (individual.sampleGuids || []).some(
-        sampleGuid => samplesByGuid[sampleGuid].isActive,
-      )),
+      [familyGuid]: familySamples.some(({ isActive }) => isActive),
     }), {})
   },
 )
 
 export const getIGVSamplesByFamilySampleIndividual = createSelector(
-  getSortedIndividualsByFamily,
+  getIndividualsByGuid,
   getIgvSamplesByGuid,
-  (individualsByFamily, igvSamplesByGuid) => {
-    return Object.entries(individualsByFamily).reduce((acc, [familyGuid, individuals]) => ({
-      ...acc,
-      [familyGuid]: individuals.reduce((familyAcc, { individualGuid, igvSampleGuids }) => {
-        (igvSampleGuids || []).forEach((sampleGuid) => {
-          const sample = igvSamplesByGuid[sampleGuid]
-          const type = sample.sampleType
-          if (!familyAcc[type]) {
-            familyAcc[type] = {}
-          }
-          familyAcc[type][individualGuid] = sample
-        })
-        return familyAcc
-      }, {}),
-    }), {})
-  },
+  (individualsByGuid, igvSamplesByGuid) =>
+    Object.values(igvSamplesByGuid).reduce((acc, sample) => {
```

Comment:
doing a single iteration through all the samples is faster than going through the family-individual groups and then iterating thorough all the individuals within that